### PR TITLE
add mouse button bindings for turning left/right

### PR DESCRIPTION
**Pull request by fabiangreffrath**

### CODE REVIEW

File: src/setup/mouse.c
Comment (turol) [inline src/setup/mouse.c:110]: Why did you move "Strafe On"? Why aren't the two new ones next to each other?

Comment (fabiangreffrath) [reply src/setup/mouse.c:110]: So they align up more nicely in the Mouse menu in chocolate-doom-setup.

Comment (fabiangreffrath) [reply src/setup/mouse.c:110]: You will note that the other menu items are also aligned like this.

Comment (turol) [reply src/setup/mouse.c:110]: I looked at the setup menu. Now "Strafe On" is on the left side, separate from "Strafe Left/Right" which are on the right side. This looks weird. Can you fix it so that they're all on the right? This requires that the right column has more items than left. I'm not sure the text screen code supports that.

Comment (fabiangreffrath) [reply src/setup/mouse.c:110]: I don't think that's possible. But we could probably have foreward+backward+strafeleft+straferight+strafeon on the left and turnleft+turnright+nextweap+prevweap on the right side.

File: src/doom/g_game.c
Comment (fabiangreffrath) [inline src/doom/g_game.c:386]: Hm, or should I rather change these to `if (gamekeydown[key_right] || mousebuttons[mousebturnright])`? Because, as it is now both keys/buttons add to the `side` and `angleturn` variables, respectively, which are of course capped to reasonable min/max values later, but anyway...

Comment (turol) [reply src/doom/g_game.c:386]: Would not doing that mean that players can do moves that original client doesn't allow like turning too fast? Probably safer to do it that way.

**Comment by AXDOOMER** [comment]
What's the use case for having mouse button bindings for turning left and right when the mouse is already a means of  turning left and right that's more accurate than buttons?

**Comment by fabiangreffrath** [comment]
I was asking the same, but there seems to be a demand for this: 
https://github.com/fabiangreffrath/crispy-doom/issues/725

**Comment by fragglet** [comment]
This seems like a niche/unusual setup. I'm a bit hesitant about cluttering up the setup UI with mostly-useless (to most people) bindings like this. My suggestion would be to either not merge this (make it a Crispy feature if necessary) or add the feature, but not include it in the setup UI - for such obscure/niche cases it's reasonable to tell people to edit a .cfg file to get what they want.

**Comment by fabiangreffrath** [comment]
> add the feature, but not include it in the setup UI

Alright, I have gone that way.

**Review by fragglet — APPROVED** [review]

**Comment by fragglet** [comment]
Thanks!

**Pull request merged**
